x-pack/plugins/apm/server/lib/helpers/create_es_client/create_infra_metrics_client/create_infra_metrics_client.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ESSearchRequest } from '@kbn/es-types';
+import { InfraPluginStart, InfraPluginSetup } from '@kbn/infra-plugin/server';
+import { ApmPluginRequestHandlerContext } from '../../../../routes/typings';
+import { getInfraMetricIndices } from '../../get_infra_metric_indices';
+
+interface InfraPlugin {
+  setup: InfraPluginSetup;
+  start: () => Promise<InfraPluginStart>;
+}
+
+export type InfraClient = Awaited<ReturnType<typeof createInfraMetricsClient>>;
+
+export async function createInfraMetricsClient({
+  infraPlugin,
+  context,
+}: {
+  infraPlugin: InfraPlugin;
+  context: ApmPluginRequestHandlerContext;
+}) {
+  return {
+    search: async <TParams extends Omit<ESSearchRequest, 'index'>>(
+      opts: TParams
+    ) => {
```

Comment:
```suggestion
    async search<TDocument, TParams extends Omit<ESSearchRequest, 'index'>>(
      opts: TParams
    ): Promise<InferSearchResponseOf<TDocument, TParams>> {
```

Reply:
`InfraMetricsSearchParams` is needed to make `size` and `track_total_hits` required.